base/shared/src/main/scala/scalaz/data/IStream.scala

```diff
@@ -0,0 +1,19 @@
+package scalaz
+package data
+
+trait IStreamModule {
+  type IStream[A]
+
+  def scons[A](a: =>A)(as: =>IStream[A]): IStream[A]
+  def uncons[A](as: =>IStream[A]): (A, IStream[A])
+}
+
+private[data] object IStreamImpl extends IStreamModule { self =>
+  type IStream[A] = Cofree[(A, ?), Unit]
```

Comment:
Why not `Cofree[Id, A]`?

Reply:
We were literally talking about that as you commented! It's `Cofree[Id, A]` now. I prefer it this way as well.